src/Support/Mutex.php

```diff
@@ -0,0 +1,89 @@
+<?php
+
+namespace RonasIT\AutoDoc\Support;
+
+use Illuminate\Support\Arr;
+use RuntimeException;
+
+class Mutex
+{
+    protected array $config;
+
+    protected const string MODE_FILE_READ_WRITE = 'c+';
+    protected const string MODE_FILE_READ = 'r';
+
+    public function __construct()
+    {
+        $this->config = config('auto-doc.paratests');
+    }
+
+    protected function readJsonFromStream($handle): ?array
```

Comment:
please order methods by the visibility criteria:
- public
- protected
- private